tests/lib/c_lib/testcase.yaml

```diff
@@ -1,12 +1,13 @@
 common:
   tags:
     - clib
-  arch_exclude: posix
+  filter: not CONFIG_NATIVE_APPLICATION
   integration_platforms:
     - mps2_an385
 tests:
   libraries.libc:
     ignore_faults: true
+    filter: not (CONFIG_ARCH_POSIX and CONFIG_EXTERNAL_LIBC)
```

Comment:
would `not CONFIG_EXTERNAL_LIBC` be sufficient here?

Reply:
Maybe.., unless somebody wanted to run this test with another external c library they provide?
(EXTERNAL_LIBC for the posix arch is a synonym to the host libC, for embedded archs, I understood it is as a "no libC provided by Zephyr" mode)
I just tried to minimize the impact of this change.
If you prefer this to say 
`    filter: not CONFIG_EXTERNAL_LIBC`
tell and I change it accordingly.